### ISSUE TITLE
Tomcat 8.5 ResourceLink not shown

### ISSUE THREAD

**Issue by claudioweiler** [psi-probe/psi-probe#1140]
Global resources are correctly shown on Data Sources, but links do not.

Resources info inside application says that all resources are misconfigured.

But application that use those links works without problems.

**Comment by claudioweiler** [comment]
It appears that Tomcat created a access validation to ResourceLinks.
I modified Tomcat85ContainerAdapter.addContextResourceLink to register access, it need some improvements in overall AbstractTomcatContainer structure.

```
  public void addContextResourceLink(Context context, List<ApplicationResource> resourceList,
      boolean contextBound) {

    NamingResourcesImpl namingResources = context.getNamingResources();
    for (ContextResourceLink link : namingResources.findResourceLinks()) {
      ApplicationResource resource = new ApplicationResource();

      logger.debug("reading resourceLink: {}", link.getName());
      resource.setApplicationName(context.getName());
      resource.setName(link.getName());
      resource.setType(link.getType());
      resource.setLinkTo(link.getGlobal());

      ResourceLinkFactory.registerGlobalResourceAccess(getGlobalNamingContext(), link.getName(), link.getGlobal());

      resourceList.add(resource);
    }
  }
```

**Comment by hazendaz** [comment]
@claudioweiler Does this work?  If so, can you add to a PR and also patch tomcat9.

**Comment by hazendaz** [comment]
@claudioweiler Lets see what you can create and go from there.  I'm not sure which piece you are waning in the resource resolver.  The fields / methods are plublic and/or static where needed already.  I may be missing exactly what you want changed for this but I'm more than happy to work with you on this.  If we need any of those changes that should be fine to accomplish this.

**Comment by claudioweiler** [comment]
Ok @hazendaz, I will make it my way, but I can't right now.

I've made a hack here just for make it work, no good to create a PR from that.

Tomcat 8.0 will need it too?

**Comment by hazendaz** [comment]
Fixed via #1176
